plugin.js

```diff
@@ -66,10 +58,33 @@ function onReqHandlerWrapper (fastify, hook) {
       if (cookieHeader) {
         fastifyReq.cookies = fastify.parseCookie(cookieHeader)
       }
+      fastifyRes[kReplySetCookies] = new Map()
       done()
     }
 }
 
+function fastifyCookieOnSendHandler (fastifyReq, fastifyRes, payload, done) {
+  if (fastifyRes[kReplySetCookies].size) {
+    let setCookie = fastifyRes.getHeader('Set-Cookie')
+
+    /* istanbul ignore else */
+    if (!setCookie) {
+      setCookie = []
+    } else if (typeof setCookie === 'string') {
+      setCookie = [setCookie]
+    }
+
+    for (const c of fastifyRes[kReplySetCookies].values()) {
```

Comment:
can you do for me one last performance test?

```suggestion
    const values = fastifyRes[kReplySetCookies].values()
    let c = ''
    for (var i = 0; i < values.length; ++i) {
      c = values[i]
```

Reply:
Can we do that?

It's yet another iterator, let me see

Reply:
make the let c without initializer. Thought c would be a string, but its a value. 

so it should be just `let c`, maybe makes sense to do `const c =` in the loop instead.

Reply:
```js
    let c = ''
    for (c of fastifyRes[kReplySetCookies].values()) {
      setCookie.push(cookie.serialize(c.name, c.value, c.opts))
    }
```

I can do this?

Reply:
I have the following thought: 

By assigning the .values() to a variable, it maybe is not a iterator anymore. Maybe by doing the .length in the loop makes it an array. Because keep in mind: it is an iterator, and if you do the for const of, it can always be that the Map changes in the loop itself. So theoretically the values() iterator has to lookup every time what the current value really is.

But if it is not an iterator, and just an array, we have no potential overhead of the iterator anymore.

Maybe it has to be

`const values = [...fastifyRes[kReplySetCookies].values()]`

to force it to be a true array.

I am just thinking, because you reported an about 10% drop in performance for single cookie.

But if everything does not result in a perf gain, than the current implementation is the way to go.

Reply:
Yeah I see, seems like it's still an iterator

MDN: The values() method returns a new [map iterator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator) object that contains the values for each element in the Map object in insertion order.

I think the difference for one cookie came from this fast path:
```js
if (!setCookie) {
    reply.header('Set-Cookie', serialized)
    return reply
  }
```

This changed to always creating an array

I will see for `const values = [...fastifyRes[kReplySetCookies].values()]`, but `...` already runs the iterator once to fill the array if I'm not mistaken

Reply:
yes, but i assume ... does it once. Maybe v8 does some optimization. Idk.

Just see, what happens. Usually a benchmark will give clarity ;).

Anyway you have my approval ;).